LEGO1/lego/legoomni/src/infocenter/infocenter.cpp

```diff
@@ -15,20 +43,135 @@ Infocenter::~Infocenter()
 // STUB: LEGO1 0x1006ed90
 MxResult Infocenter::Create(MxDSAction& p_dsAction)
 {
-	return FAILURE;
+	OutputDebugString("create called\n");
+	if (LegoWorld::Create(p_dsAction) == SUCCESS) {
+		InputManager()->SetWorld(this);
+		ControlManager()->Register(this);
+	}
+
+	LegoGameState* gs = GameState();
+	m_infocenterState = (InfocenterState*) gs->GetState("InfocenterState");
+	if (!m_infocenterState) {
+		m_infocenterState = (InfocenterState*) gs->CreateState("InfocenterState");
+		m_infocenterState->SetUnknown0x74(3);
+	}
+	else {
+		// TODO
+	}
+
+	// TODO
+	InputManager()->Register(this);
+	SetIsWorldActive(FALSE);
+	return SUCCESS;
 }
 
-// STUB: LEGO1 0x1006ef10
+// FUNCTION: LEGO1 0x1006ef10
 MxLong Infocenter::Notify(MxParam& p_param)
 {
-	// TODO
+	LegoWorld::Notify(p_param);
+
+	if (m_worldStarted) {
+
+		switch (((MxNotificationParam&) p_param).GetNotification()) {
+		case 0:
+			return HandleNotification0(p_param);
+		case c_notificationEndAction:
+			return HandleEndAction(p_param);
+		case c_notificationKeyPress:
+			return HandleKeyPress(((LegoEventNotificationParam&) p_param).GetKey()) & 0xff;
```

Comment:
Typically, & 0xff bit masks (at least from what I know) suggests types of u8's (MxU8)